libs/wingsdk/src/target-tf-gcp/table.ts

```diff
@@ -0,0 +1,76 @@
+import { Construct } from "constructs";
+import {
+  BigtableTable,
+  BigtableTableConfig,
+  BigtableTableColumnFamily,
+} from "../.gen/providers/google/bigtable-table";
+import { Id } from "../.gen/providers/random/id";
+import * as ex from "../ex";
+import {
+  ResourceNames,
+  NameOptions,
+  CaseConventions,
+} from "../shared/resource-names";
+import { Json } from "../std";
+
+/*
+ * Table names must be between 1 and 50 characters. We reserve 9 characters for
+ * a random ID, so the maximum length is 41.
+ *
+ * Must be 1-50 characters and must only contain
+ * hyphens, underscores, periods, letters and numbers.
+ *
+ * We skip generating a hash since we need to append a random string to the
+ * bucket name to make it globally unique.
+ *
+ */
+const TABLE_NAME_OPTS: NameOptions = {
+  maxLen: 41,
+  case: CaseConventions.LOWERCASE,
+  disallowedRegex: /([^a-z0-9.\-\_]+)/g,
+  includeHash: false,
+};
+
+const INSTANCE_NAME_OPTS: NameOptions = {
+  maxLen: 30,
+  disallowedRegex: /()/g,
+};
+
+/**
+ * GCP implementation of `ex.Table`.
+ *
+ * @inflight `@winglang/sdk.ex.ITableClient`
+ */
+export class Table extends ex.Table {
+  constructor(scope: Construct, id: string, props: ex.TableProps = {}) {
+    super(scope, id, props);
+
+    const tableName = ResourceNames.generateName(this, TABLE_NAME_OPTS);
+    const instanceName = ResourceNames.generateName(this, INSTANCE_NAME_OPTS);
+
+    const randomId = new Id(this, "Id", {
+      byteLength: 4, // 4 bytes = 8 hex characters
+    });
+
+    let columnsFamily: BigtableTableColumnFamily[] = [];
+    for (let key in this.columns) {
+      columnsFamily.push({ family: key });
+    }
+
+    const config: BigtableTableConfig = {
+      name: tableName + `-${randomId.hex}`,
+      instanceName: instanceName,
+      columnFamily: columnsFamily,
+    };
+
+    new BigtableTable(this, "Default", config);
+  }
+
+  public addRow(key: string, row: Json): void {
+    throw new Error(`Can not insert ${row} at given ${key}`);
```

Comment:
As far as I can understand, it is impossible to implement this function at this point, since it requires a client.

reference: https://cloud.google.com/bigtable/docs/samples/bigtable-insert-update-rows